### DISCUSSION TITLE
Is there support for DPI LCD displays with HAT?

### DISCUSSION THREAD

**Discussion by Preciseful** [rsta2/circle#474] [Q&A] [answered]
More specifically, is there any support for displays like [this](https://www.waveshare.com/wiki/7inch_LCD_for_Pi) with an RGB LCD HAT?
I really wanna be able to use it, and I'm thinking maybe the raspberry pi touch monitor would be compatible, but I haven't messed around with it.

**Comment by rsta2** [comment]
I'm unsure. I tend to say, it does not work, because it needs device-tree overlays. It will not work with the Raspberry Pi 5 in any case.

**Reply by Preciseful** [reply]
So there's a very slim chance it would properly work on a RPI4B?

**Reply by rsta2** [reply, marked as answer]
I did more reading and unfortunately now I think, it will not work. Circle totally relies on the firmware for HDMI and DSI display support. After reading [this](https://forums.raspberrypi.com/viewtopic.php?p=2211174#p2211246) I assume, there is no DPI support in the firmware and hence, it won't work. Newer Raspberry Pi OS versions use kernel mode drivers for display support (not the firmware), but such a driver is not available for Circle.

**Reply by Preciseful** [reply]
So my only option left is to make my own driver? (which I assume is definitely not gonna be easy)

**Reply by rsta2** [reply]
No, it's not easy. The Raspberry Pi OS kernel mode display support is [here](https://github.com/raspberrypi/linux/tree/rpi-6.6.y/drivers/gpu/drm/vc4). [This](https://docs.kernel.org/gpu/vc4.html) can be of interest too.

**Reply by Preciseful** [reply]
Well if I ever get to around doing it and I manage it, I'll try to make a PR.

**Reply by rsta2** [reply]
Yes, it's difficult. But if you manage it, I'm looking forward to your PR.
